src/main/java/com/jwplayer/southpaw/state/RocksDBState.java

```diff
@@ -336,9 +336,10 @@ public void configure(Map<String, Object> config) {
             this.restoreMode = RestoreMode.parse(config.getOrDefault(RESTORE_MODE_CONFIG, RestoreMode.NEVER.getValue()).toString());
             this.uri = new URI(Preconditions.checkNotNull(config.get(URI_CONFIG).toString()));
 
+            // TODO: We need to split S3 persistence out of RocksDBState. It doesn't belong here
```

Comment:
agreed. the state class should not care if its S3, GFS or the local file system.  This would be a welcome abstraction